File: shifai-ios/ShifAI/App/DeepLinkRouter.swift
import Foundation
import SwiftUI

/// Deep Link Router — handles shifai:// URL scheme.
/// Registered in Info.plist CFBundleURLSchemes.
final class DeepLinkRouter: ObservableObject {

    enum Destination: Equatable {
        case dashboard
        case tracking
        case insights
        case export
        case settings
        case syncConflict
        case onboarding
        case unknown
    }

    @Published var activeDestination: Destination?

    /// Parse incoming deep link URL and route.
    /// Supported URLs:
    /// - shifai://dashboard
    /// - shifai://tracking
    /// - shifai://insights
    /// - shifai://export
    /// - shifai://settings
    /// - shifai://sync/conflict
    /// - shifai://auth/callback?token=xxx
    func handle(_ url: URL) {
        guard url.scheme == "shifai" else { return }

        let host = url.host ?? ""
        let path = url.path

        switch host {
        case "dashboard":
            activeDestination = .dashboard
        case "tracking":
            activeDestination = .tracking
        case "insights":
            activeDestination = .insights
        case "export":
            activeDestination = .export
        case "settings":
            activeDestination = .settings
        case "sync":
            if path == "/conflict" {
                activeDestination = .syncConflict
            }
        case "auth":
            handleAuthCallback(url)
        case "app":
            activeDestination = .dashboard // Default landing
        default:
            activeDestination = .unknown
        }
    }

    /// Handle OAuth callback from Supabase Auth
    private func handleAuthCallback(_ url: URL) {
        guard let components = URLComponents(url: url, resolvingAgainstBaseURL: false),
              let token = components.queryItems?.first(where: { $0.name == "token" })?.value else {
            return
        }
        // Store token and navigate
        Task {
            await SupabaseClient.shared.setAccessToken(token)
        }
        activeDestination = .dashboard
    }

    func clearDestination() {
        activeDestination = nil
    }
}

File: shifai-android/app/src/main/java/com/shifai/presentation/navigation/DeepLinkRouter.kt
package com.shifai.presentation.navigation

import android.content.Intent
import android.net.Uri

/**
 * Deep Link Router — handles shifai:// URL scheme.
 * Registered in AndroidManifest.xml intent-filter.
 */
object DeepLinkRouter {

    enum class Destination {
        DASHBOARD, TRACKING, INSIGHTS, EXPORT, SETTINGS, SYNC_CONFLICT, AUTH_CALLBACK, UNKNOWN
    }

    data class RouteResult(
        val destination: Destination,
        val params: Map<String, String> = emptyMap()
    )

    /**
     * Parse incoming deep link URI and return routing info.
     * Supported URIs:
     * - shifai://dashboard
     * - shifai://tracking
     * - shifai://insights
     * - shifai://export
     * - shifai://settings
     * - shifai://sync/conflict
     * - shifai://auth/callback?token=xxx
     * - shifai://app (default → dashboard)
     */
    fun parse(uri: Uri): RouteResult {
        if (uri.scheme != "shifai") return RouteResult(Destination.UNKNOWN)

        val host = uri.host ?: ""
        val path = uri.path ?: ""

        return when (host) {
            "dashboard" -> RouteResult(Destination.DASHBOARD)
            "tracking" -> RouteResult(Destination.TRACKING)
            "insights" -> RouteResult(Destination.INSIGHTS)
            "export" -> RouteResult(Destination.EXPORT)
            "settings" -> RouteResult(Destination.SETTINGS)
            "sync" -> {
                if (path == "/conflict") RouteResult(Destination.SYNC_CONFLICT)
                else RouteResult(Destination.UNKNOWN)
            }
            "auth" -> {
                val token = uri.getQueryParameter("token")
                RouteResult(Destination.AUTH_CALLBACK, buildMap {
                    token?.let { put("token", it) }
                })
            }
            "app" -> RouteResult(Destination.DASHBOARD)
            else -> RouteResult(Destination.UNKNOWN)
        }
    }

    fun parseFromIntent(intent: Intent): RouteResult? {
        val uri = intent.data ?: return null
        return parse(uri)
    }

    /** Navigation route names matching Compose NavHost */
    fun destinationToRoute(destination: Destination): String = when (destination) {
        Destination.DASHBOARD -> "dashboard"
        Destination.TRACKING -> "tracking"
        Destination.INSIGHTS -> "insights"
        Destination.EXPORT -> "export"
        Destination.SETTINGS -> "settings"
        Destination.SYNC_CONFLICT -> "sync_conflict"
        Destination.AUTH_CALLBACK -> "dashboard"
        Destination.UNKNOWN -> "dashboard"
    }
}
